lib/Order.rb

```diff
@@ -0,0 +1,46 @@
+require_relative 'menu.rb'
+
+# This new class is going to create an order
+class Order
+
+  attr_reader :dishes
+
+  def initialize
+    @dishes = []
+    @menu = Menu.new
+    @complete = false
+  end
+
+  def add_dish(name, amount = 1)
+    @name = name
+    @amount = amount
+    fail "This dish isn't available, please run menu.list_dishes to see the dishes" unless available
+
+    @dishes << { :name => name, :amount => amount }
```

Comment:
Great use of arrays and hashes, really clear and simple

Reply:
If you use a symbol as a hash key you can use this syntax

`{name: name, amount: amount}`

It's more idiomatic that way.